### PR TITLE
Updating Regress2sum

### PR DESCRIPTION
**Pull request by ymt123**
Several cleanup updates to regression code:

- Break out model definitions to separate files
- Allow for either MSE or Negative Sampling loss functions
- Several items to speed up iteration including on GPU embedding lookup and dataset in memory loading
- Modify evaluation code to return values instead of printing

### CODE REVIEW

File: attalos/imgtxt_algorithms/regress2sum/regress2sum.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def train_model(train_dataset,
     # Extract features and place in numpy matrix
     for i in test_dataset:
         image_feats, tags = test_dataset[i]
-        val_image_feats[i, :] = image_feats/np.linalg.norm(image_feats)
+        val_image_feats[i, :] = image_feats
```
Comment (agude) [inline attalos/imgtxt_algorithms/regress2sum/regress2sum.py]: Was removing the normalization intentional?

Comment (ymt123) [inline attalos/imgtxt_algorithms/regress2sum/regress2sum.py]: Yes, I had removed it from the training and failed to remove it from the test data. I found that it did not improve performance.

**Comment by ymt123** [comment]
This PR requires #87 to work properly (Tensorflow 10)

**Comment by agude** [comment]
I get this runtime error:

```
Traceback (most recent call last):
  File "attalos/imgtxt_algorithms/regress2sum/regress2sum.py", line 318, in <module>
    main()
  File "attalos/imgtxt_algorithms/regress2sum/regress2sum.py", line 292, in main
    for i, line in enumerate(input_file):
UnboundLocalError: local variable 'input_file' referenced before assignment
```

But it looks like that should always be defined.

**Comment by ymt123** [comment]
Added fix for that error (which occurs when an invalid path is specified for the word vector file)

**Pull request merged**